### ISSUE TITLE
double disable error

### ISSUE THREAD

**Issue by JiangWeixian** [stylelint/stylelint#5069]
> no-duplicate-selectors has already been disabled (CssSyntaxError)

e.g. "There are false positives for two leading zeros when ..."

```ts
const lint = await stylelint
  .lint({
    syntax,
    config: {
      rules: {}
    },
    code: content
  })
  .then(res => res);
```

```console
{
  errored: true,
  results: [
    {
      source: "<input css 1>",
      deprecations: [],
      invalidOptionWarnings: [],
      parseErrors: [],
      errored: true,
      warnings: [Array]
    }
  ],
  output:
    '[{"source":"<input css 1>","deprecations":[],"invalidOptionWarnings":[],"parseErrors":[],"errored":true,"warnings":[{"line":4,"column":1,"rule":"CssSyntaxError","severity":"error","text":"\\"no-duplicate-selectors\\" has already been disabled (CssSyntaxError)"}]}]',
  reportedDisables: []
};
```

> Which rule, if any, is the bug related to?

e.g. `number-leading-zero`

> What code is needed to reproduce the bug?

e.g.

```css
/* stylelint-disable at-rule-empty-line-before,at-rule-name-space-after,at-rule-no-unknown */
/* stylelint-disable no-duplicate-selectors */
/* stylelint-disable */
/* stylelint-disable declaration-bang-space-before,no-duplicate-selectors,string-no-newline */
/* stylelint-disable at-rule-no-unknown */
html,
body {
  width: 100%;
  height: 100%;
}
input::-ms-clear,
input::-ms-reveal {
  display: none;
}
```

> Which version of stylelint are you using?

`13.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

`Nodejs API`

**Comment by jeddy3** [comment]
@JiangWeixian Thanks for the report.

I believe its working as intended. The `no-duplicate-selectors` rule is being disabled twice. You should write:

```css
/* stylelint-disable at-rule-empty-line-before,at-rule-name-space-after,at-rule-no-unknown */
/* stylelint-disable no-duplicate-selectors, string-no-newline, declaration-bang-space-before  */
html,
body {
  width: 100%;
  height: 100%;
}
input::-ms-clear,
input::-ms-reveal {
  display: none;
}
```

See [this demo](https://stylelint.io/demo#N4Igxg9gJgpiBcID0AqABAZwC4E8A2MeAlgHZYC0URGAhgEYFo0UBOArgeSROWyQNbcA7iTQokAHRKpMuAsTKVq9Rs3LtO3XgOGjxIADQgMOMjQAeCcBgyHwEEgDMiAcyvApaNBJAaYGH3g0D1EvbxA1Py4ePkEIEUC0LHYYA08wnwYIMH5o8hgAWwAHXETktlT0rx9IPAgWPNIANxpiKHIACxhzMpS00Or7AoKYRS1Ckpxeiv6w8NgwPBoWZiIHciycvKg2IuIwZhhyIpYIIpgWLCJ-RIBtKrDyyoGwkLm5n1duFhg7h-fwpASBgYGA2FcmkcdnsiAcsP5yEIiFgOkpHI4LqMKC08BUAiB-mEALqEgC+DyJswyIAWSxWV3Wm1yWgwHXqKJoJHaJzOF1w5AgkJYLCIsHxQSeVMGjgcFEcNAKRDwOG2u32hy4CpuCCSfQePhlinliuVeUVNlILnILlGF1h5GNSpV-BgOCE9Sg02eHxAjj4YAZJHIBzwYEaJBxoq9UvCfpIAbWQZDYa0fAwRRoYBg7V59Pq0f1vv9gfIChgy2tKyoWLy3GBWE5UGW7SoPwTDgLLx8LpwjhWI0ooLpq3WWiIxXZnKwnZ9IyoNAd5awbB+moHqZ08RIM+pWjFWa5lvI6dBRGcYGRUx1ksLe7VsI1anHRXZ6g42oleq7IDvMLhRxBAgA3qcVdRmW8eAmfkMAgFcsx3QZxnMZIFxBRVagcUCb2-McI1aUVKFghgAKWVlgwgYYsSwr8fR5c5LhVDc4gSa8aOpQDQSwepjhBNgoB4RYaBsPJYl0BDwg44CGiKXj+PyAgRjGGJNxYz9wO-SSuIaXBzhElTt1Y9SfWwEUSCtLQSBgIQy3Enw+GRPTmIMtTvWpSJ33yYp+TLDYYBlH4-heQZWiEGgcHxGMvDeAFwm6LMSkCmLqSZAhhJoRx4QaWh1y1DY6hyVKIsJH1nBYbAuH8eFPQJILiUi6kvnqX4dXuWqfXSzLyMosgfGKtASVq8kXkpQt3M0XKDhBRIfDqIQLh8GMfDGo4SFy9NMyODr5p1Ja8FC8Lj0tTgbMMUbWA8tDYQgOog0s6zSE2jLtqCXb9qK78mWDOoMCPOgVizTzJlLB7fP85qXp-GAhQWwtPsWCAfrMjZ-pWqyfK2lhpoiPawven04e+36UYq+7LNBpqsZC3HyAKDgrmByyYY+-LcnhxGrT+jbjwzAG6D8imduxt7DrM46HqZ-GWYFc4SCJrm7vRp7McFqmDtpvB6ZOxaQE+3lZaRzmAfWgGMcpnGDvZsXGdO5nslyPW5aNnmjj5sGzbeiXqQwhounMYMhPB8JZu27Xvc6bpS1GFwUSx1l2U9wZIG6igoJVHzXYFoJWoBV7cYT14+p8OKYASlq+ulIgyooSzsGzXq2v6+rBkagKy4bwZsHwQhSAoJOCkbfE+oGgEhrmEbvz7mshA6ZF-Gd8hSB+2B3bzm2fTBbAKOOU56P5VOGZd-nW6zwlc-C-OosLkBi9Lk-2-CDHgzYTeCm33NSjXpLY0r8qa6q+ukrjySp8Fw3xA7Zy-ktJWXVFLTk-sAkAi9RQASOkcdOLMAExWHu8UedVCy0mWCOIMdBORWmNo9TKWNLLQ3gfMIchCSwkKRuQ8mx8H7mzxtSAh9JEx5TtseQoV0bok0VpQlWHCaZ0yIAffOPhuFEL4VsS6GEgwsNNuI4Wls0Hi1oXI+hPDGRS2UdddYLCM5sJ8NQkO+D9EKM+log+NMLCDkEgY4EiQACM2t5GMKlihJUR5jE3RXufXRNJbElhUSIkG6iIaqwwJIjW0itY2NcQoqJailYhISQ4lJ34fG8IyfPcxgdLFQ2sfkiJvD97pyPuA0+QtV71Wijna+5h4pwLvl-B+0CCkdloa00qv9Kp1xqoApu4QW71PvlAzqk8eoDPeJ8YEyCRYuCti7DBYysFkgpNrOM7YkwUX7tEsmsT2HC3VprHR+ziyFOOahee5yz45NQTIsJBzIkPO5lzEpVDynK1ufGEs-c-Y1IeqBAADECw5a4jiTVKSAYOgLCyfN4RmH4ZAugggSQrEGSDl4aOplc5JNzUV3PWBirE2KEQsIJYiqxWU3l5JKhSoM09Z6ZLEXEjhsjEFcixEQxIAAmbWBAzIonIAALwuDwDcyJbIgFBYDbyELPHaznEQBcGJmArnhSY1RTyslEtCRq7MWrFy6tXEU35dT-k0LNfOS1y5VyrQHAirGyK+Wau1UuPVxxljUv8LS+e9L7UVNnOa31VqjgrDMkcXMzBuJcuehcpphYfXOv9XGm00sLhJqysUu1JrOGDEzQARwqCwNO1Be7fLxWc41PLLlSPeY6i1laLg1vKn3R5XNnmNItsysl34K1Vu7XW4YfbebFohoyvl4xrqIhnvCchiqtDmnZvkLkApHDHlgiwAGDbA7YR9CQNgBQ+YNAIDQKgSMZWnGyQui9V68j+IUFaB9CNFV0T5CqD1gsvVhKiOCsmfy26tPiYk651tmlXxvp0tAECEFDOriM6qQ8JkgLAYlSBERoHzLgUPXZw1taaW4swZIRA6Dghdv9F0WAcmhpWYSudAK+XkYaJRkUNH4R5rzIW-tTbwjzrCZx8g3HqO0f4wWn5M63aC1E2RwgnFuJJzoKQWTKblbNvTRpFTUkuoadWlpOTh8FO6dNYWcT+5RhNkUOpzTpmLKmOdoq8TxAe31rRjE4TLyOMGdM55ydJyzGzpE+xsTgXuLKtAx+NA0LrPRekrJASpEEkAYhkB5TQFTMyRgHxNLQkElUqxcGpjXMw2Kcizl1TKWCtyRU7A-2U1APxAjexZLPEGuQQUjWKJ3AGyBixvxNgxEAu5e4jpeFAdPXtZRd+eyvdZttbmgtn0OIKjkB7O6Fg7RMtB3m3yzb2ivNTtORQ1NLzoOktg4WE7wMzuhaNdytNQ7RbaLu9+B7wWYHTvM5nCLDr7utC2792LXka010SJC9IQ1SQgFJEAA):

i.e. the following will trigger the same error.

```css
/* stylelint-disable at-rule-no-unknown */
/* stylelint-disable at-rule-no-unknown */
```

We probably shouldn't classify it as a `CssSyntaxError`, though, as that's confusing.

**Comment by JiangWeixian** [comment]
@jeddy3 thanks

```ts
/* stylelint-disable at-rule-no-unknown */
/* stylelint-disable at-rule-no-unknown */
```

But, I'm a little confused about why double disable ignore will trigger error...(not just classify problem)

**Comment by jeddy3** [comment]
I suspect the original design decision was to surface overlapping or invalid disables ranges to the user because [ranges can get complicated](https://github.com/stylelint/stylelint/blob/master/lib/__tests__/disableRanges.test.js) and debugging them would be a pain.

**Comment by niklasHagner** [comment]
+1

It's weird that a comment is categorized as a CssSyntaxError. It get that it might be seen as an error but it really should have a separate class.

I've been looking into using styleLint to validate that my compiled css files are correct, but since I'm building multiple sass-files into one css-file, that css-file can contain duplicates of 'stylelint-disable' comments.

If the error thrown was a separate type other than CssSyntaxError I guess it could be easier to filter the SassLint error output and detect if there are any real css errors
